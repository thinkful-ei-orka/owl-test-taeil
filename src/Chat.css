.chat {
  position: relative;
}
.chat-log {
  height: calc(100% - 55px);
  overflow-y: auto;
}
.chat-input {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  padding: 10px;
}
.chat-input input {
  width: 100%;
  height: 45px;
  font-size: 16px;
  padding: 10px;
  color: #333;
}
.message {
  display: flex;
}
.message img {
  width: 40px;
  height: 40px;
  background: #ddd;
  border-radius: 50%;
  margin-right: 10px;
}
.message div {

}
.message h6 {
  margin: 0;
  font-size: 16px;
  margin-top: 3px;
}
.message .time {
  color: #adadad;
  font-weight: regular;
  font-size: 14px;
}
.chat-action {
  color: #adadad;
  text-align: center;
  margin: 15px;
  font-size: 14px;
}
