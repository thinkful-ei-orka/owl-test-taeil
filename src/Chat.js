import React from 'react';

import './Chat.css';
import data from './data';

function Chat() {
  let chatEvents = [];

  data.chatEvents.forEach((chatEvent, index) => {
    let participant = data.participants.filter((participant) => participant.id === chatEvent.participantId)[0];
    switch (chatEvent.type) {
      case 'message':
        let time = new Date(chatEvent.time);
        let hh = time.getHours();
        let m = time.getMinutes();
        let ampm = 'AM';
        if (hh >= 12) {
          hh = hh - 12;
          ampm = "PM";
        }
        if (m.toString().length === 1) {
          m = '0' + m.toString();
        }
        chatEvents.push(
          <div className='message' key={index}>
            <img src={participant.avatar} alt={participant.name + ' avatar'} />
            <div>
              <h6>{participant.name} <span className="time">{hh + ':' + m + ' ' + ampm}</span></h6>
              <div>{chatEvent.message}</div>
            </div>
          </div>
        );
        break;
      case 'thumbs-up':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> gave a thumbs up <i className="far fa-thumbs-up"></i></div>);
        break;
      case 'thumbs-down':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> gave a thumbs down <i className="far fa-thumbs-down"></i></div>);
        break;
      case 'raise-hand':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> raised their hand <i className="far fa-hand-paper"></i></div>);
        break;
      case 'clap':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> clapped <i className="fas fa-hands-wash"></i></div>);
        break;
      case 'join':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> joined</div>);
        break;
      case 'leave':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> left</div>);
        break;
      case 'join-stage':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> joined the stage</div>);
        break;
      case 'leave-stage':
        chatEvents.push(<div className="chat-action" key={index}><strong>{participant.name}</strong> left the stage</div>);
        break;
      default:
        break;
    }
  });

  return (
    <section className='chat'>
      <div className="chat-log">
      {chatEvents}
      </div>
      <div className="chat-input"><input type="text" placeholder="Chat"></input></div>
    </section>
  );
}

export default Chat;
