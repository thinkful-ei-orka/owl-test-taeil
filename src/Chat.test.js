import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Chat from './Chat'

describe(`Chat Component`, () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    expect(
      renderer.create(<Chat />).toJSON()
    ).toMatchSnapshot()
  })
})
