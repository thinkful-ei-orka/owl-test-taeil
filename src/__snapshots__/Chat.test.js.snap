// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Component renders the UI as expected 1`] = `
<section
  className="chat"
>
  <div
    className="chat-log"
  >
    <div
      className="message"
    >
      <img
        alt="Axe Kubicka avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <div>
        <h6>
          Axe Kubicka
           
          <span
            className="time"
          >
            6:50 AM
          </span>
        </h6>
        <div>
          Hello world
        </div>
      </div>
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Caty Flucker
      </strong>
       gave a thumbs up 
      <i
        className="far fa-thumbs-up"
      />
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Frank Runciman
      </strong>
       gave a thumbs down 
      <i
        className="far fa-thumbs-down"
      />
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Koren Templeton
      </strong>
       raised their hand 
      <i
        className="far fa-hand-paper"
      />
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Koren Templeton
      </strong>
       clapped 
      <i
        className="fas fa-hands-wash"
      />
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Koren Templeton
      </strong>
       joined
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Ashla Attwool
      </strong>
       left
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Axe Kubicka
      </strong>
       joined the stage
    </div>
    <div
      className="chat-action"
    >
      <strong>
        Axe Kubicka
      </strong>
       left the stage
    </div>
  </div>
  <div
    className="chat-input"
  >
    <input
      placeholder="Chat"
      type="text"
    />
  </div>
</section>
`;
